lib/haml/railtie.rb

```diff
@@ -1,22 +1,19 @@
-if defined?(ActiveSupport)
-  require 'haml/template/options'
-  ActiveSupport.on_load(:before_initialize) do
-    ActiveSupport.on_load(:action_view) do
-      require "haml/template"
-    end
-  end
-end
+require 'haml/template/options'
 
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+
+      ActiveSupport.on_load(:action_view) do
```

Comment:
This is the recommended approach as documented here: http://pothibo.com/2015/4/customize-rails-view-controller-with-railtie and here http://blog.carbonfive.com/2011/02/25/configure-your-gem-the-rails-way-with-railtie/

In any case, the important bit is the `on_load`, otherwise the bug is not fixed. Currently the initializer runs too early.